### ISSUE TITLE
more official mapping for "roleTerm should be ignored if namePart doesn't have a value" ?

### ISSUE THREAD

**Issue by ndushay** [sul-dlss-labs/cocina-descriptive-metadata#92]
In addressing sul-dlss/dor-services-app/issues/1161, my goal was to avoid getting a cocina mapping error for a round trip with the below in mods, from pd967mn2579 on stage (https://argo-stage.stanford.edu/view/druid:pd967mn2579).  

```xml
          <name>
            <namePart/>
            <role>
              <roleTerm authority="marcreleator" type="text"/>
            </role>
          </name>
```

I translated the above into 

```json
          {
            "name": [
              {
                "value": ''
              }
            ],
            "role": [
              {
                "source":
                  {
                    "code": 'marcreleator'
                  }
              }
            ]
          }
```

This achieved 2 goals
- avoids the "key not found" error for the round trip from fedora mods xml to cocina-models back to mods
- gives us a roundtrip with as much useful information as the original:

```xml
            <name>
              <namePart/>
              <role/>
            </name>
```

I confess that I didn't want to muck about with this edge case further, so it's not an exact match on the roundtrip.

I also confess I punted on the "et al" case from https://github.com/sul-dlss/dor-services-app/issues/1161#issuecomment-709565285:

> roleTerm should be ignored if namePart doesn't have a value (unless etal, a sibling tag to roleTerm is present, which is an edge case)

Justin thinks this APO was created by Hydrus, and thus may not be a complete anomaly, and suggests we should either get the proper mapping from you or ... ????

**Comment by arcadiafalcone** [comment]
The role-without-a-name is an unfortunate quirk of how Hydrus generates MODS. It would be OK (and in fact preferable) to drop the name-role entry in the round trip back to MODS.

**Comment by arcadiafalcone** [comment]
Closed by #95 and #100
